oddt/fingerprints.py

```diff
@@ -0,0 +1,184 @@
+
+"""
+Module checks interactions between two molecules.
+"""
+
+import numpy as np
+import oddt
+from oddt.interactions import close_contacts, pi_stacking, pi_cation, \
+    hbond_acceptor_donor, salt_bridge_plus_minus, hydrophobic_contacts, acceptor_metal
+
+__all__ = ['InteractionFingerprint', 'SimpleInteractionFingerprint']
+
+
+def InteractionFingerprint(protein, ligand, strict=True):
+    """
+        Interaction fingerprint accomplished by converting the molecular
+        interaction of ligand-protein into bit array according to
+        the residue of choice and the interaction. For every residue
+        (One row = one residue) there are eight bits which represent
+        seven type of interactions:
+
+        - (Column 0) hydrophobic contacts
+        - (Column 1) aromatic face to face
+        - (Column 2) aromatic edge to face
+        - (Column 3) hydrogen bond (protein as hydrogen bond donor)
+        - (Column 4) hydrogen bond (protein as hydrogen bond acceptor)
+        - (Column 5) salt bridges (protein positively charged)
+        - (Column 6) salt bridges (protein negatively charged)
+        - (Column 7) salt bridges (ionic bond with metal ion)
+
+        Parameters
+        ----------
+        protein, ligand : oddt.toolkit.Molecule object
+            Molecules to check interactions beetwen them
+        strict : (deafult = True)
+            If True, include condition, which informs whether atoms
+            form 'strict' H-bond (pass all angular cutoffs).
+            If strict is None do not include angular condintion.
+
+        Returns
+        -------
+        InteractionFingerprint : numpy array
+            Vector of calculated IFP (size = no residues * 8 type of interaction)
+
+    """
+    resids = np.unique(protein.atom_dict['resid'])
+    IFP = np.zeros((len(resids), 8), dtype=np.uint8)
+
+    # hydrophobic contacts (column = 0)
+    hydrophobic = hydrophobic_contacts(protein, ligand)[0]['resid']
+    np.add.at(IFP, [np.searchsorted(resids, hydrophobic), 0], 1)
+
+    # aromatic face to face (Column = 1)
+    aromatic = pi_stacking(protein, ligand)[0]['resid']
+    np.add.at(IFP, [np.searchsorted(resids, aromatic), 1], 1)
+
+    # aromatic edge to face (Column = 2)
+    aromatic_1 = pi_cation(protein, ligand)[0]['resid']
+    np.add.at(IFP, [np.searchsorted(resids, aromatic_1), 2], 1)
+
+    # h-bonds, protein as a donor (Column = 3)
+    _, donors, strict0 = hbond_acceptor_donor(ligand, protein)
+    if strict is True:
+        strict = strict0
+    else:
+        strict = None
+    np.add.at(IFP, [np.searchsorted(resids, donors[strict]['resid']), 3], 1)
+
+    # h-bonds, protein as an acceptor (Column = 4)
+    acceptors, _, strict1 = hbond_acceptor_donor(protein, ligand)
+    if strict is not None:
+        strict = strict1
```

Comment:
After I gave some thoughts, it should be the other way around - if `strict` is `False`, then strict1 = None and use `strict1` in next line. Otherwise we overwrite strict and next `if` statement will be broken if we try to do not strict SIFT.